internal/services/resource/resource_deployment_script_common.go

```diff
@@ -0,0 +1,606 @@
+	AzureCliKind        DeploymentScriptKind = "AzureCli"
+)
+
+type ResourceDeploymentScriptModel struct {
+	Name                   string                             `tfschema:"name"`
+	ResourceGroupName      string                             `tfschema:"resource_group_name"`
+	Arguments              string                             `tfschema:"arguments"`
+	Version                string                             `tfschema:"version"`
+	CleanupPreference      deploymentscripts.CleanupOptions   `tfschema:"cleanup_preference"`
+	ContainerSettings      []ContainerConfigurationModel      `tfschema:"container"`
+	EnvironmentVariables   []EnvironmentVariableModel         `tfschema:"environment_variable"`
+	ForceUpdateTag         string                             `tfschema:"force_update_tag"`
+	Location               string                             `tfschema:"location"`
+	PrimaryScriptUri       string                             `tfschema:"primary_script_uri"`
+	RetentionInterval      string                             `tfschema:"retention_interval"`
+	ScriptContent          string                             `tfschema:"script_content"`
+	StorageAccountSettings []StorageAccountConfigurationModel `tfschema:"storage_account"`
+	SupportingScriptUris   []string                           `tfschema:"supporting_script_uris"`
+	Tags                   map[string]string                  `tfschema:"tags"`
+	Timeout                string                             `tfschema:"timeout"`
+	Outputs                string                             `tfschema:"outputs"`
+}
+
+type ContainerConfigurationModel struct {
+	ContainerGroupName string `tfschema:"container_group_name"`
+}
+
+type EnvironmentVariableModel struct {
+	Name        string `tfschema:"name"`
+	SecureValue string `tfschema:"secure_value"`
+	Value       string `tfschema:"value"`
+}
+
+type StorageAccountConfigurationModel struct {
+	StorageAccountKey  string `tfschema:"key"`
+	StorageAccountName string `tfschema:"name"`
+}
+
+type ResourceDeploymentScriptPatchModel struct {
+	Tags map[string]string `tfschema:"tags"`
+}
+
+func getDeploymentScriptArguments(kind DeploymentScriptKind) map[string]*pluginsdk.Schema {
+	result := map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9_()-.]{0,259}[a-zA-Z0-9_()-]$`),
+				`The name length must be from 1 to 260 characters. The name can only contain alphanumeric, underscore, parentheses, hyphen and period, and it cannot end with a period.`,
+			),
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"retention_interval": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ISO8601DurationBetween("PT1H", "P1DT2H"),
+		},
+
+		"arguments": {
```

Comment:
Just to call out again that I think this should be `command_line` to make it more clear what this attribute is referring to